src/Adapter/Routing/AdminLinkBuilder.php

```diff
@@ -0,0 +1,101 @@
+<?php
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Routing;
+
+use Link;
+use PrestaShop\PrestaShop\Core\Routing\EntityLinkBuilderInterface;
+
+/**
+ * Class AdminLinkBuilder is able to build entity links based on the Link::getAdminLink
+ * method (which indirectly allows it to build symfony url as well).
+ */
+class AdminLinkBuilder implements EntityLinkBuilderInterface
+{
+    /** @var Link */
+    private $link;
+
+    /** @var array */
+    private $entitiesControllers;
+
+    /**
+     * @param Link $link
+     * @param array $entitiesControllers
```

Comment:
Please add more details regarding the schema for this parameter. I should be able to learn how use any method without needing to look up usage examples elsewhere.

Reply:
You mean details in the PHPDoc or strong typed objects?

Reply:
I meant a detailed PHPdoc, thanks